CMakeLists.txt

```diff
@@ -277,18 +277,32 @@ set(CPACK_COMPONENT_UNSPECIFIED_DEPENDS resources)
 set(CPACK_COMPONENT_DEMO_DEPENDS Unspecified)
 
 if (APPLE)
-   # see the following URL for information about these variables
-   # https://developer.apple.com/library/mac/#documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
-   set(BUNDLE_CFBundleShortVersionString ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_STAGE})
-   set(BUNDLE_CFBundleVersion ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_STAGE})
+  # see the following URL for information about these variables
+  # https://developer.apple.com/library/mac/#documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
+  set(BUNDLE_CFBundleShortVersionString ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_STAGE})
+  set(BUNDLE_CFBundleVersion ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_STAGE})
 
-   set(CPACK_PACKAGE_ICON "${RESOURCEDIR}/GrandOrgue.icns")
+  set(CPACK_PACKAGE_ICON "${RESOURCEDIR}/GrandOrgue.icns")
 
-   configure_file(${CMAKE_SOURCE_DIR}/src/grandorgue/resource/Info.plist.in ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.app/Contents/Info.plist)
-   INSTALL(FILES ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.app/Contents/Info.plist DESTINATION "${PROJECT_NAME}.app/Contents/")
+  configure_file(${CMAKE_SOURCE_DIR}/src/grandorgue/resource/Info.plist.in ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.app/Contents/Info.plist)
+  INSTALL(FILES ${CMAKE_BINARY_DIR}/${PROJECT_NAME}.app/Contents/Info.plist DESTINATION "${PROJECT_NAME}.app/Contents/")
 
-  set (CPACK_SYSTEM_NAME "osx")
+  set (CPACK_SYSTEM_NAME "macOS")
   set (CPACK_GENERATOR DragNDrop)
+	
+	# script to codesign the app on the .dmg, executed before building the .dmg
+	file(WRITE "${CMAKE_BINARY_DIR}/sign_package.cmake"
```

Comment:
1. Could you make this script static and put it to the `cmake` subdirectory?
2. Maybe someone wants to do full signing instead of the ad-hoc one. Could you provide cmake variables to switch this?

Reply:
2. Sorry but this is beyond my Cmake and code sign knowledge and I can't test full signing.